src/hooks/UrqlUseMutation.re

```diff
@@ -34,16 +34,25 @@ let useMutation = (~request) => {
   let parse = request##parse;
 
   let (stateJs, executeMutationJs) = useMutationJs(query);
+  let hasExecuted = React.useRef(false);
 
   let state =
     React.useMemo2(
-      () => UrqlResponse.urqlResponseToReason(~response=stateJs, ~parse),
+      () =>
+        UrqlResponse.urqlResponseToReason(
+          ~response=stateJs,
+          ~parse,
+          ~hasExecuted=hasExecuted->React.Ref.current,
+        ),
       (stateJs, parse),
     );
 
   let executeMutation =
     React.useMemo2(
       ((), ~context=?, ()) => {
+        if (!hasExecuted->React.Ref.current) {
+          hasExecuted->React.Ref.setCurrent(true);
```

Comment:
Once we call `executeMutation` for the first time, we set `hasExecuted` to `true`. If we get no data or errors on next render, we'd match the `Empty` constructor.